### ISSUE TITLE
false genotyping large deletion

### ISSUE THREAD

**Issue by DHmeduni** [fritzsedlazeck/Sniffles#505]
Hi, I have a 62kb deletion that is very obviously heterzygous but sniffles is calling it as homozygous even the coverage field its's obvious its heterozygous..... anybody have an idea what to do, or is this a bug?

chr7	148238576	Sniffles2.DEL.4A5S6	N	<DEL>	60	PASS	PRECISE;SVTYPE=DEL;SVLEN=-62659;END=148301235;SUPPORT=83;COVERAGE=**154,75,57,79,157**;STRAND=+-;AF=1.000;STDEV_LEN=0.000;STDEV_POS=0.000	GT:GQ:DR:DV	1/1:60:0:83

**Comment by mletexier-cnrgh** [comment]
Hi @DHmeduni & @fritzsedlazeck ,
I have the same phenomenon, which is repeated on several events, for example, a 39 Kb deletion, which is seen with the 1/1 genotype with sniffles (VAF=100%, (0, 25)) whereas visually in IGV, it is 0/1. Sequences in IGV are sorted according to haplotypes. 
I've attached the IGV screenshot. 
The parameters used are: 
--minsupport 5 
--tandem-repeats hs38me.trf.bed 
 --minsvlen 50 
--min-alignment-length 2000 
--mapq 20

![FalseGenotype_DEL_IGV](https://github.com/user-attachments/assets/430d9a72-92fc-4ee1-a30c-338a046cb4cc)

Thanks for any help you can give
Mélanie

**Comment by fritzsedlazeck** [comment]
Thanks we can reproduce this now also here.. @lfpaulin is working on that. Hope we have a fix on the next verison. 

We found the issue , which is basically handeling the read counting wrong for large split read events.. which is a bug in the deletions but works on all other SV events.  In short we are by far undercounting the ref supporting reads , which leads to a 1/1 GT. 

Thanks
Fritz

**Comment by mletexier-cnrgh** [comment]
Thank you very much for your prompt feedback, Fritz. 
Looking forward to the next version :)

Mélanie

**Comment by tuannguyen8390** [comment]
Hi @fritzsedlazeck! Do you have any estimate for when the next release might be coming? We're really looking forward to seeing this fix as this having an impact on our recessive lethal tracking 🍄

**Comment by fritzsedlazeck** [comment]
We have a working prototype that goes through testing right now. It looks like we resolved it. We hope that we can roll out that new version next week. 
Thanks
Fritz

**Comment by tuannguyen8390** [comment]
Thanks Fritz, really looking forward to it :) 

Tuan

**Comment by tuannguyen8390** [comment]
Sorry for having a back-to-back message Fritz, but would the bug cause error in the existing snf files or can we just run the merge SNF with the new Sniffles version ?

**Comment by fritzsedlazeck** [comment]
So this is still something we are investigating. @lfpaulin can maybe expand a bit more... the SNF file holds all variants but they have set filters.. These filters can be overwritten with new behavior during the merge, which also applies the same filters.

**Comment by tuannguyen8390** [comment]
We test out v.2.5 today. Seems like we need to re-generate the old snf file 

1. call from bam
```
zgrep 91857666 4780_2.5.2.vcf.gz                                                                              

9       91857666        Sniffles2.DEL.E0CS8     N       <DEL>   60      PASS    PRECISE;SVTYPE=DEL;SVLEN=-138347;END=91996012;SUPPORT=15;COVERAGE=34,20,11,13,27;STRAND=+-;STDEV_LEN=0.000;STDEV_POS=0.000;VAF=0.500        GT:GQ:DR:DV     0/1:60:15:15
```

2. call from 2.5 snf
```
zgrep 91857666 4780_snf_2.5.2_new_snf.vcf.gz                                                                      
9       91857666        Sniffles2.DEL.2210M8    N       <DEL>   60      PASS    PRECISE;SVTYPE=DEL;SVLEN=-138347;END=91996012;SUPPORT=15;COVERAGE=34,20,11,13,27;STRAND=+-;STDEV_LEN=0.000;STDEV_POS=0.000;VAF=0.500        GT:GQ:DR:DV:ID  0/1:60:15:15:Sniffles2.DEL.E0CS8
```

3. call from 2.2 snf
```
zgrep 91857666 4780_snf_2.5.2.vcf.gz

9       91857666        Sniffles2.DEL.2210M8    N       <DEL>   60      PASS    PRECISE;SVTYPE=DEL;SVLEN=-138347;END=91996012;SUPPORT=15;COVERAGE=34,20,11,13,27;STRAND=+-;AF=1.000;STDEV_LEN=0.000;STDEV_POS=0.000 GT:GQ:DR:DV:ID  1/1:41:0:15:Sniffles2.DEL.E0CS8
```

**Comment by fritzsedlazeck** [comment]
Sorry for that. I can cross check with the team if there is a mode to rescue form the SNF file.. the SV should be in there but has already a failed tag assigned. @lfpaulin knows this better than I by now ..
Thanks
Fritz

**Comment by tuannguyen8390** [comment]
Thanks @fritzsedlazeck , would be v. nice if we can have that option. Asking 17 partners to remake their snf is a bit of a nightmare to me 😆.

**Comment by fritzsedlazeck** [comment]
Yeah... I also have a similar situation. Meeting with @hermannromanek next monday and will discuss. Stay tune pls. 
Fritz

**Comment by tuannguyen8390** [comment]
Hi Hermann,

Just to reconfirm stuff,

- Case 2 & 3, yes, we were having wrong genotype using v.2.5 to call SV for SNF generated from v2.2. I understand that even it's single sample call, under our testing we have the same calling phenomenon with multiple samples. 
- Yes, the 9:91857666 is a known SV on bovine that we know it should not be 1/1. As per my previous comment v2.5 call it correctly as 0/1 from v2.5 SNF, while called it 1/1 from v2.2 SNF. 

I can share the 2.2 snf to you. Uploading now. Can I drop you an email for confidentiality?

Cheers,

Tuan

**Comment by hermannromanek** [comment]
Hi Tuan,

Thank you for confirming - I've added some code allowing sniffles to re-genotype variants from old SNF files and we'll check with human samples but I also wanted to verify it fixes your issue, or do some more analysis if it doesn't. You can reach me at sniffles@romanek.at

Thanks for your help,
Hermann

**Comment by tuannguyen8390** [comment]
Hi @hermannromanek ,

Email sent, please let me know if there is any problem accessing files

Tuan

**Comment by fritzsedlazeck** [comment]
@hermannromanek can we close this or is there still something we need to change ?
Thanks
Fritz
